sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs

```diff
@@ -6,21 +6,29 @@ pub(crate) mod stronghold_snapshot;
 use std::{fs, path::PathBuf, sync::atomic::Ordering};
 
 use futures::{future::try_join_all, FutureExt};
-use zeroize::Zeroize;
 
 use self::stronghold_snapshot::{read_data_from_stronghold_snapshot, store_data_to_stronghold};
 #[cfg(feature = "storage")]
 use crate::wallet::WalletBuilder;
 use crate::{
-    client::secret::{stronghold::StrongholdSecretManager, SecretManager, SecretManagerDto},
+    client::{
+        secret::{stronghold::StrongholdSecretManager, SecretManager, SecretManagerDto},
+        utils::Password,
+    },
     types::block::address::Hrp,
     wallet::{Account, Wallet},
 };
 
 impl Wallet {
     /// Backup the wallet data in a Stronghold file
     /// stronghold_password must be the current one when Stronghold is used as SecretManager.
-    pub async fn backup(&self, backup_path: PathBuf, mut stronghold_password: String) -> crate::wallet::Result<()> {
+    pub async fn backup(
+        &self,
+        backup_path: PathBuf,
+        stronghold_password: impl Into<Password> + Send,
+    ) -> crate::wallet::Result<()> {
+        let stronghold_password = stronghold_password.into();
```

Comment:
Same here?

Reply:
can you make it a suggestion as well? then I'll commit it as a batch through GH.

Reply:
No, I don't want to have it added :trollface: